include/swift/SILOptimizer/Utils/ValueLifetime.h

```diff
@@ -54,24 +55,47 @@ class ValueLifetimeAnalysis {
   /// end the value's lifetime.
   using Frontier = SmallVector<SILInstruction *, 4>;
 
+  /// A type erased version of frontier so callers can customize the inline
+  /// size.
+  using FrontierImpl = SmallVectorImpl<SILInstruction *>;
+
   /// Constructor for the value \p def with a specific range of users.
   ///
   /// We templatize over the RangeTy so that we can initialize
   /// ValueLifetimeAnalysis with misc iterators including transform
   /// iterators.
   template <typename RangeTy>
-  ValueLifetimeAnalysis(decltype(defValue) def, const RangeTy &userRange)
-      : defValue(def), userSet(userRange.begin(), userRange.end()) {
+  ValueLifetimeAnalysis(SILArgument *def, const RangeTy &useRange)
```

Comment:
Why is this renamed? `useRange` should have Operand types, but this still expects instruction pointers, or is there supposed to be a call to `getUser` here?

Also, why do we require a list of use Operands if we're only storing the set of user instructions?